### DISCUSSION TITLE
How many requests the api can respond to per second？

### DISCUSSION THREAD

**Discussion by emiao0** [osm-search/Nominatim#2992] [Q&A] [answered]
I have installed a local version of Nominatim with 14G RAM, 8 cores CPU and 1T SSD( import the entire planet data with "admin" style) . And it is able to process requests at about 20 times per second. Is this speed reasonable? 

If I want to achieve an order of magnitude increase in speed, is this possible? (because my data volume is a bit large...) would I need to use a machine with more RAM in the first place?

Thanks!

**Comment by mtmail** [comment, marked as answer]
20 should be possible on 8 core CPU. On https://nominatim.org/release-docs/latest/admin/Deployment/ look at the `php-fpm` software. php-fpm also works with Apache not just nginx. It allows to run many parallel processes that keep listening and processing incoming requests in an event-queue and is superior to starting and stopping processes. You'll need to experiment with the settings (https://www.php.net/manual/en/install.fpm.configuration.php) on how many processes should to match your number of CPU cores.

More RAM will always help, then Postgres can cache more indices. For full planet you'll need 32GB for the import, 64GB is better. That's for the osm2pgsql step which is in the first 2-3 hours of the import so you can try with 14GB first. You can reduce RAM after the import finished.

**Reply by emiao0** [reply]
Thank you so much! I will try php-fom first !

But I still wonder if nominatim is possible to reach 200 or even 2000 requests per second, and what resources(CPU/SSD/...) are needed. Because this speed would be helpful for my research, thank you!

**Reply by lonvia** [reply]
200 requests/s are possible but will already need a big machine (128GB RAM, 16 cores, NVME disks). Our oldest production machine [dulcy](https://hardware.openstreetmap.org/servers/dulcy.openstreetmap.org/) peaks at around 200 req/s. With even more hardware you can possibly get to 1000 req/s. I had our new machine [longma](https://hardware.openstreetmap.org/servers/longma.openstreetmap.org/) do 600 requests/s and it still was far away from peak load. But as you see in the spec, it means doubling the hardware once more.

In the end it also depends what kinds of requests you want to do. /reverse is better than /search by at least a factor of 2. The numbers above are for a mix of reverse and search requests.
